### DISCUSSION TITLE
Bad Request / HTTP 400 Error

### DISCUSSION THREAD

**Discussion by luke-ransfield** [Budibase/budibase#6348] [Help] [answered]
Have been testing Budibase over the last 48hrs, suddenly all apps are returning `Bad Request` toasts and HTTP 400 Errors in the dev console. This includes the Budibase template apps as well as the apps I've been working on. In Firefox there is a HTTP 404 warning as well, as a warning about the site not having the same `samesite` attribute value?

`Cookie “budibase:currentapp” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite [agency-client-portal](https://xframe.budibase.app/app/agency-client-portal)`

`[Client] HTTP 404 on POST:https://xframe.budibase.app/api/bbtel/ping
	Not Found [budibase-client.js:243:222465](https://cdn.budi.live/app_xframe_d123ee7cff8540508a5096b756e5f920/budibase-client.js?v=1.0.199)`

`[Client] HTTP 400 on POST:https://xframe.budibase.app/api/ta_d97bef4247e34e85b4f54efa626b02e4/rows
	Bad Request`

Any reason why this is happening?

**Comment by melohagan** [comment]
Hi @lukxfnz 

Are you self hosting? If so, have you tried doing a `docker-compose down` followed by an `up` ?

If using the CLI, could be worth doing an upgrade as well: https://docs.budibase.com/docs/budibase-cli-setup#updating-budibase

**Reply by luke-ransfield** [reply]
Hey @melohagan, thanks for speedy reply!

Not self-hosting, and have been using BB 's inbuilt database. Not sure if that makes debugging easier or more complicated...

**Comment by melohagan** [comment]
Maybe try clearing your cookies and logging in again?

**Reply by luke-ransfield** [reply]
Hey @melohagan, have tried in a private browser session and using Chrome (instead of FF) and both didn't fix the issue.

**Comment by melohagan** [comment, marked as answer]
For this error here: `[Client] HTTP 400 on POST:https://xframe.budibase.app/api/ta_d97bef4247e34e85b4f54efa626b02e4/rows Bad Request`

Could you please open the network tab and share the more detailed response error message please

**Comment by luke-ransfield** [comment]
Hey @melohagan ! Thanks for suggesting to view the more detailed error response; rather embarrassingly I had changed the data-structure in BB to include more required inputs, but had not updated the form-updater to include those fields. The 400 error was coming back because some of the required fields were empty.

 Might be a nice idea to have the data-repeater auto-update the form to include the new fields if the database changes?

**Reply by melohagan** [reply]
No worries! That's actually come up a few times, so I've raised a bug here: https://github.com/Budibase/budibase/issues/6356

Your idea is great too, but would require more work, so probably won't happen for a while.
